File: 71-simplify-path/71-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> stack = new Stack<>();
        String[] parts = path.split("/");
        for (String part: parts) {
            if (part.equals("..")) {
                if (!stack.empty()) {
                    stack.pop();
                }
            } else if (part.equals(".") || part.equals("")) {
                continue;
            } else {
                stack.push(part);
            }
        }
        
        String result = "";
        Iterator s = stack.iterator();
        while (s.hasNext()) {
            result += "/" + s.next();
        }
        if (result.equals("")) {
            result = "/";
        }
        return result;
    }
}

File: 71-simplify-path/71-simplify-path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        paths = path.split("/")
        stack = []
        for path in paths:
            if path == "..":
                if len(stack) != 0:
                    stack.pop()
            elif path == ".":
                continue
            elif path == "":
                continue
            else:
                stack.append(path)
                
        result = "/"
        for i in range(len(stack)):
            result += stack[i]
            if i != len(stack) - 1:
                result += "/"
            
        return result